src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ProcessorTestSuiteForOcrValidation.java

```diff
@@ -0,0 +1,191 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.tasks;
+
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.web.client.HttpServerErrorException;
+import org.testcontainers.containers.DockerComposeContainer;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.bulkscanprocessor.config.ContainerMappings;
+import uk.gov.hmcts.reform.bulkscanprocessor.config.IntegrationTest;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.Envelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.EnvelopeRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.ProcessEventRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.ocrvalidation.client.OcrValidationClient;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.servicebus.ServiceBusHelper;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.storage.OcrValidationRetryManager;
+import uk.gov.hmcts.reform.bulkscanprocessor.util.TestStorageHelper;
+import uk.gov.hmcts.reform.bulkscanprocessor.validation.OcrPresenceValidator;
+import uk.gov.hmcts.reform.bulkscanprocessor.validation.OcrValidator;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+@IntegrationTest
+@TestPropertySource(properties = {
+    "scheduling.task.scan.enabled=true"
+})
+abstract class ProcessorTestSuiteForOcrValidation {
```

Comment:
Would it be possible to reuse existing ProcessorTestSuite? Now there is some duplication, and similar code to be maintained in two places. 

If not maybe similar methods should be extracted into the helpers, i.e.uploadToBlobStorage and similar.

Reply:
Implemented `ProcessorTestSuiteForOcrValidation -> ProcessorTestSuite` inheritance. Getting rid fully of `ProcessorTestSuiteForOcrValidation` is too cumbersome for this task.